### PR TITLE
docs: Update AutoComplete first demo with control mode & add faq

### LINKED ISSUE #18230: antd v3.10.10升级到v3.21.2导致该问题。AutoComplete受控时，使用onSearch监控输入事件，不能输入中文。
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/embed/autocomplete-bug-recurrent-odfuk)

### Steps to reproduce
1、AutoComplete受控
2、使用onSearch监控输入事件
3、输入中文

### What is expected?
期望能输入中文

### What is actually happening?
不能输入中文

| Environment | Info |
|---|---|
| antd | 3.21.2 |
| React | 16.8.6 |
| System | macOS 10.12.6 |
| Browser | chrome75.0.3770.142 |

---
虽然可以用onChange事件代替onSearch事件，但是不管怎样，不能输入中文(还有日文、韩文IME等)，影响还是挺大的。

追根溯源，该问题是由于使用了setState function updater导致的。
react-component/select/blob/master/src/Select.tsx  -> setInputValue():
```js
const { onSearch } = this.props;
if (inputValue !== this.state.inputValue) {
      this.setState(prevState => {
        // Additional check if `inputValue` changed in latest state.
        if (fireSearch && inputValue !== prevState.inputValue && onSearch) {
          onSearch(inputValue);
        }
        return { inputValue };
      }, this.forcePopupAlign);
    }
```
react-component/select/blob/master/src/Select.tsx  -> getDerivedStateFromProps():
```js
if ('value' in nextProps) {
      const value = Select.getValueFromProps(nextProps);
      newState.value = value;
      if (nextProps.combobox) {
        newState.inputValue = Select.getInputValueForCombobox(nextProps, optionsInfo);
      }
    }
```
解读:
setState function updater里调用onSearch，当前子组件和父组件生命周期调用顺序如下：
当前子组件render -> 父组件render -> 当前子组件render，因为当前子组件先render一遍，而inputValue是从Props获取值的，inputValue会被设置两次值，并且第一次设置会把IMF最后一个输入字符去掉，这样input value会被设置两次不同的值，导致不能输入中文。

经测试，
onSearch在setState里面调用，会导致当前子组件render两次，父组件render一次；
onSearch在setState外面调用，当前子组件、父组件都只会render一次。所以onSearch放在setState外面调用也没关系。

我写了个小例子，input元素不能输入中文，重现上面的情况。
[https://codesandbox.io/embed/disable-imf-z2zkj](https://codesandbox.io/embed/disable-imf-z2zkj)

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #18230

-----
[View rendered components/auto-complete/demo/basic.md](https://github.com/ant-design/ant-design/blob/docs/components/auto-complete/demo/basic.md)
[View rendered components/auto-complete/index.en-US.md](https://github.com/ant-design/ant-design/blob/docs/components/auto-complete/index.en-US.md)
[View rendered components/auto-complete/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/docs/components/auto-complete/index.zh-CN.md)

### CODE REVIEW

File: components/auto-complete/demo/basic.md [hunk 1]
```diff
@@ -22,25 +22,41 @@ function onSelect(value) {
...
+          onSearch={this.onSearch}
+          placeholder="input here"
+        />
+        <br />
+        <br />
+        <AutoComplete
+          dataSource={dataSource}
+          style={{ width: 200 }}
+          onSelect={onSelect}
+          onSearch={this.onSearch}
```
Comment (afc163) [inline components/auto-complete/demo/basic.md]: onChange?

Comment (afc163) [reply components/auto-complete/demo/basic.md]: 受控模式应该有 `value` 和 `onChange`。

Comment (zombieJ) [reply components/auto-complete/demo/basic.md]: 噗，checkout 把 demo 还原了 😈

Comment (zombieJ) [reply components/auto-complete/demo/basic.md]: done

Comment (afc163) [reply components/auto-complete/demo/basic.md]: 要不要拆开，这个 `<Complete />` 改成 `<ControlledComplete />`。

**Review by afc163 — APPROVED** [review]

**Pull request merged**